### PR TITLE
Drop c style api

### PR DESCRIPTION
**Pull request by tylershunt**
Barrier doesn't seem useful anymore (and probably never was).

### CODE REVIEW

File: libtsuba/include/tsuba/tsuba.h [hunk 1]
```diff
@@ -1,29 +1,71 @@
...
+void Fini();
+
+/* Download a file and open it */
+int Open(const std::string& uri);
+
+/* map a particular chunk of this file (partial download)
+ * @begin and @size should be well aligned to TSUBA_BLOCK_SIZE
+ * return value will be aligned to tsuba_block_size as well
+ */
+uint8_t* Mmap(const std::string& filename, uint64_t begin, uint64_t size);
```
Comment (witchel) [inline libtsuba/include/tsuba/tsuba.h:43]: Probably too detailed, but why is this called Mmap?  Do we really want to support partial downloads?  Do we have to remember that we only downloaded part of this file when we try to write it back?

In general, I'm getting the vibe that the storage layer should be in charge of memory management when it relates to buffering files.  That isn't so compatable with an mmap-like interface.

File: libgalois/src/FileView.cpp [hunk 2]
```diff
@@ -15,36 +15,36 @@ FileView::~FileView() { Unbind(); }
...
   }
 }
 
 int FileView::Bind(const std::string& filename) {
-  TsubaStatBuf buf;
-  int err = TsubaStat(filename.c_str(), &buf);
-  if (err) {
+  tsuba::StatBuf buf;
+  int err = tsuba::Stat(filename, &buf);
+  if (!err) {
```
Comment (ddn0) [inline libgalois/src/FileView.cpp:26]: `if (err)`; this is already fixed in master.

File: libtsuba/include/tsuba/tsuba.h (hunk 1 above)
Comment (ddn0) [reply libtsuba/include/tsuba/tsuba.h:43]: FWIW, my view is that in the limit we probably have to bubble up buffering to the application layer (like how MPI has blocking and nonblocking sends), but in the short-term, most applications will be a single bulk-load, compute and then single bulk-store.

How this shakes out in terms of an interface and its evolution is another story.

File: libgalois/src/FileView.cpp (hunk 2 above)
Comment (tylershunt) [reply libgalois/src/FileView.cpp]: Yeah sloppy merge my bad.

File: libtsuba/include/tsuba/tsuba.h (hunk 1 above)
Comment (tylershunt) [reply libtsuba/include/tsuba/tsuba.h:43]: Yeah Mmap isn't the right name here, but I expect the api to evolve a lot so I'm going to hold off on optimizing it.

**Pull request merged**